### ISSUE TITLE
Feature request: pin proto version

### ISSUE THREAD

**Issue by W1M0R** [moonrepo/proto#547]
It would be nice to be able to pin proto on a per-project basis using the `.prototools` file of the project, e.g.

project A:
```toml
proto = "0.38.1"
```

project B:
```toml
proto = "0.38.2"
```

The usefulness of this, is that one version of `proto` might bring a backwards-incompatible change (on purpose or by mistake): perhaps in the way the `.prototools` file is processed or the `.prototools` tree resolution method or some plugin defaults etc.

I imagine that the user would always have the latest version of `proto` installed globally, but when proto is executed in the context of a `.prototools` file, it will respect the configuration, download the older version of proto and continue execution using that version of proto. The shims should probably be able to handle this situation.

This would also allow more freedom to implement potentially breaking changes (i.e. new defaults) in newer versions of proto, but still have confidence that users who have pinned their proto version in prototools will not be affected by the new defaults until they opt-in to the new version. This has its limits, but could be useful.

I'm not sure if a TOML plugin called proto would be sufficient for this, but perhaps a built-in WASM plugin could handle this. If on the plugin layer, then pinning to an older version of proto would also use the older proto plugin to interpret the pinning behaviour. I'm not sure what would be the best approach.

This could potentially also be implemented outside the plugin framework, as a core setting, e.g. 

```toml
[settings]
proto_version = "=0.38.2"
```

**Comment by milesj** [comment]
This is actually how moon works. It downloads a versioned copy of proto to `~/.proto/tools/proto` and only runs with that explicit version. It should be easy enough to also update proto to do the same.

<img width="920" alt="Screenshot 2024-07-15 at 9 28 47 AM" src="https://github.com/user-attachments/assets/16c10069-1871-47d4-aef1-58f964a6f000">

**Comment by milesj** [comment]
Landed in v0.39.
